python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -351,6 +363,20 @@ def _apply_default_remote_args(ray_remote_args: Dict[str, Any]) -> Dict[str, Any
     def get_autoscaling_actor_pools(self) -> List[AutoscalingActorPool]:
         return [self._actor_pool]
 
+    def update_resource_usage(self) -> None:
+        """Updates resources usage."""
+        actors = list(self._actor_pool._num_tasks_in_flight.keys())
+        for actor in actors:
+            actor_state = actor._get_local_state()
+            if actor_state != gcs_pb2.ActorTableData.ActorState.ALIVE:
+                # If an actor is not ALIVE, it's a candidate to be marked as a
+                # restarting actor.
+                self._actor_pool.running_to_restarting(actor, actor.get_location)
```

Comment:
```suggestion
                if self._actor_pool.is_actor_running(actor):
                    self._actor_pool.running_to_restarting(actor, actor.get_location.remote())
```
* Moving the running check here would be cleaner. And more importantly, we should only send `get_location` when the actor switched from running to restarting. 
* `.remote()` was missed after `get_location`. It didn't error out probably because actor locality is disabled by default right now.

Reply:
Ah ok. get_location is valid only when state is ALIVE